src/index.tsx

```diff
@@ -171,12 +172,30 @@ const SUPPORTS_PATH2D = (function () {
 })();
 
 function QRCodeCanvas(props: QRPropsCanvas) {
+  const {
+    value,
+    size = DEFAULT_SIZE,
+    level = DEFAULT_LEVEL,
+    bgColor = DEFAULT_BGCOLOR,
+    fgColor = DEFAULT_FGCOLOR,
+    includeMargin = DEFAULT_INCLUDEMARGIN,
+    style,
+    imageSettings,
+    ...otherProps
+  } = props;
+  const imgSrc = imageSettings?.src;
   const _canvas = useRef<HTMLCanvasElement>(null);
   const _image = useRef<HTMLImageElement>(null);
 
-  function update() {
-    const {value, size, level, bgColor, fgColor, includeMargin} = props;
+  // We're just using this state to trigger rerenders when images load. We
+  // Don't actually read the value anywhere. A smarter use of useEffect would
+  // depend on this value.
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  const [isImgLoaded, setIsImageLoaded] = useState(false);
```

Comment:
This would get better in a future version. I wanted behavior to remain the same in 3.x (always update) but in 4.x we'd use this as an input to `useEffect`